### PR TITLE
Update Brexit CTA on contextual sidebar

### PR DESCRIPTION
**Pull request by alex-ju**
## What
Update branded Brexit sidebar navigation placement, visual and logic

## Why
To drive engagement through a more targeted approach

## Visual Changes

[Preview component](https://govuk-publis-update-bre-vsk8qa.herokuapp.com/component-guide/contextual_sidebar/with_brexit_cta/preview)

<table>
<tr><th>Before</th><th>After</th></tr>
<tr><td valign="top">

localhost_3212_component-guide_contextual_sidebar_default_preview (1)

</td><td valign="top">

localhost_3212_component-guide_contextual_sidebar_default_preview

</td></tr>
</table>

## AT testing
<table>
<tr><th>AT</th><th>Result</th></tr>
<tr><td valign="top">JAWS on Chrome</td><td valign="top">

Brexit campaign slogan. Check, change, go. Check what you need to do. link

</td></tr>
<tr><td valign="top">NVDA on Firefox</td><td valign="top">

Brexit campaign slogan. Check, change, go. Check what you need to do. link

</td></tr>
<tr><td valign="top">VoiceOver on iOS</td><td valign="top">

Link, Brexit campaign slogan, Check, Change, Go. Check what you need to do.

</td></tr>
<tr><td valign="top">High Contrast on Edge</td><td valign="top">

Windows High Contrast Edge

</td></tr>
<tr><td valign="top">Custom colours on Firefox</td><td valign="top">

Firefox custom colours

</td></tr>
</table>

[Trello card](https://trello.com/c/XmaPJL11)

### CODE REVIEW

File: lib/govuk_publishing_components/presenters/contextual_navigation.rb [hunk 1]
```diff
@@ -114,7 +114,77 @@ def show_brexit_related_links?
...
+          fa13521f-9285-45b0-bd65-4a472a8037e7
+          0fab9131-f877-4286-b4d8-922fbfb402b6
+          08a8a69f-2825-4fe2-a4cf-c83458a5629e
+          3b4d6319-fcef-4637-b35a-e3df76321894
+          6426d1c5-93c8-4659-85d5-1f0d3368a124
+        ]
+
+        document_exceptions.include?(content_item["content_id"]) ||
+          document_type_exceptions.include?(content_item["document_type"]) ||
+          content_item.dig("links", "taxons").to_a.any? { |taxon| taxon_exceptions.include?(taxon["content_id"]) && !taxon_allow_list.include?(taxon["content_id"]) }
```
Comment (sihugh) [inline lib/govuk_publishing_components/presenters/contextual_navigation.rb:187]: Should either or both of the taxon_allow_list/taxon_exceptions lists take into account the full tree of taxons, and not just the ones they're directly tagged to?

For example, https://www.gov.uk/api/content/government/news/transport-secretary-launches-794-million-investment-to-boost-rail-links-in-north-and-south is tagged to `transport/rail-infrastructure`, and therefore indirectly `transport`. This won't be reflected here unless we traverse down the tree of linked taxons.

File: app/views/govuk_publishing_components/components/_contextual_sidebar.html.erb [hunk 2]
```diff
@@ -33,4 +31,8 @@
   <% if navigation.show_brexit_related_links? && navigation.step_by_step_count > 0 %>
     <%= render 'govuk_publishing_components/components/contextual_sidebar/brexit_related_links' %>
   <% end %>
+
+  <% if !navigation.show_brexit_related_links? && navigation.step_by_step_count.zero? && !navigation.transition_countdown_exception? %>
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/views/govuk_publishing_components/components/_contextual_sidebar.html.erb]: We needed it before too, but it was covered by the fact that we were in an `elsif` branch of `if navigation.show_brexit_related_links?`, hence we knew that `!navigation.show_brexit_related_links?` was `true`. 

Extracted the diff, hopefully that makes a bit more sense – it wasn't the most explicit conditional statement, indeed.

```diff
  <% if navigation.show_brexit_related_links? && navigation.step_by_step_count.zero? %>
    <%= render 'govuk_publishing_components/components/contextual_sidebar/brexit_related_links' %>
- <% elsif navigation.step_by_step_count.zero? && !navigation.transition_countdown_exception? %>
-   <%= render 'govuk_publishing_components/components/contextual_sidebar/brexit_cta' %>
- <% end %>

+ <% if !navigation.show_brexit_related_links? && navigation.step_by_step_count.zero? && !navigation.transition_countdown_exception? %>
+   <%= render 'govuk_publishing_components/components/contextual_sidebar/brexit_cta' %>
+ <% end %>
```

To sum up, we only show the custom Brexit transition element if:
- we don't already show the brexit related link (as in this case we use the [standard related links](https://github.com/alphagov/govuk_publishing_components/pull/1888/files/4d0f3c79f65dc96cc024f3534c2059ef25b77ef4#diff-20e8d7d7f967163605038425325bf11fef8644dd0cb039619628951e678e0f3cL33-L35))
- we don't have a step-by-step
- and there is no exception flagged by the `brexit_cta_exception` helper

File: lib/govuk_publishing_components/presenters/contextual_navigation.rb [hunk 3]
```diff
@@ -114,7 +114,77 @@ def show_brexit_related_links?
       end
 
       def transition_countdown_exception?
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply lib/govuk_publishing_components/presenters/contextual_navigation.rb]: Yes, I will break it up – was a bit concerned about spreading this exception-related logic into functions that we might use somewhere else or expand on and then make it harder to clean up the logic – but it does make it easier to follow though.

File: lib/govuk_publishing_components/presenters/contextual_navigation.rb (hunk 1 above)
Comment (alex-ju) [reply lib/govuk_publishing_components/presenters/contextual_navigation.rb:187]: Good point, Simon. Will update to look at the full taxon tree.

File: app/views/govuk_publishing_components/components/_contextual_sidebar.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-ju) [reply app/views/govuk_publishing_components/components/_contextual_sidebar.html.erb]: Refactored the code and tried to simplify the logic in this and moved into the presenter. We now only have a `navigation.show_brexit_cta?` that wraps the partial render.

**Review by sihugh — APPROVED** [review]
Thanks Alex 👍

**Pull request merged**